### PR TITLE
Add check pull request reviews action

### PR DESCRIPTION
**Pull request by adcreare**
Closes https://github.com/checkdigit/hash/issues/15
Added two reviewers. 
Action should run after the first review and then fail until the second reviewer approves..

### CODE REVIEW

**Review by carlansley — APPROVED** [review]

**Comment by carlansley** [comment]
@adcreare it worked, but I had to make that check "required" in the branch protection rules

**Comment by carlansley** [comment]
Actually I don't think it worked.. it failed with this error which I assume isn't right:
`Run checkdigit/github-actions/check-pr-reviews@prepare-beta
Action Error - exit 1 - error: Error: incorrect action configuration
    at haveAllReviewersReviewed (/home/runner/work/_actions/checkdigit/github-actions/prepare-beta/build/github-api/index.js:96:15)
    at reviewedCorrectly (/home/runner/work/_actions/checkdigit/github-actions/prepare-beta/build/github-api/index.js:164:40)
    at main (/home/runner/work/_actions/checkdigit/github-actions/prepare-beta/build/check-pr-reviews/index.js:15:61)
    at Object.<anonymous> (/home/runner/work/_actions/checkdigit/github-actions/prepare-beta/build/check-pr-reviews/index.js:21:1)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47`

**Comment by adcreare** [comment]
> @adcreare it worked, but I had to make that check "required" in the branch protection rules

Its actually failed somewhere (I mean it was supposed to fail, but it failed with incorrect action configuration) - thankfully i can run it again so I'll sort that out.

**Comment by carlansley** [comment]
@adcreare maybe it needs to make a comment (e.g. 1/2 reviewers approved).  or some way of providing feedback

**Comment by adcreare** [comment]
For some reason GITHUB_TOKEN is not being passed to the action. I think this might be because I'm using the `pull_request_review` hook vs the more normal ones. 
I've updated the action configuration here to push that environment var and I've added additional logging on the action - maybe give it another review and see what happens. 

Once that is sorted out I'll add in the posting of a feedback about number of reviewers left.

**Review by carlansley — APPROVED** [review]

**Comment by carlansley** [comment]
the logs look good? but didn't bomb the check

**Comment by adcreare** [comment]
Urgh - https://github.com/checkdigit/hash/actions/runs/5183818967 even has the annotation that the check did not pass correctly, but seems thats not enough for the action to fail in the view here. 
I'm going to have to convert that to an exception being thrown i guess instead of setError().

**Comment by adcreare** [comment]
Ok issued sorted out and comments added it. I'll wait for Luke to review to make sure the logic for all reviewers have approved works correctly.

**Review by lukebrdn — DISMISSED** [review]
lgtm

**Comment by adcreare** [comment]
> ✅ PR review status - All reviews completed and approved!

^ Just for you @lukebrdn  :)

**Comment by adcreare** [comment]
Ready for another look - just moved to main on the actions now the github-actions pr has been merged

**Review by carlansley — CHANGES_REQUESTED** [review]
Just realized the ci.yml doesn't actually execute any tests

File: .github/workflows/publish-beta.yml
Comment (carlansley) [inline .github/workflows/publish-beta.yml:1]: Might be worth adding the comment action here?

**Review by carlansley — CHANGES_REQUESTED** [review]

**Comment by adcreare** [comment]
Done - ready for another look

**Review by carlansley — APPROVED** [review]

**Review by lukebrdn — APPROVED** [review]
lgtm

**Review by samelliottdlt — APPROVED** [review]

**Pull request merged**